packages/core/test/unit/query-generator/select-query.test.ts

```diff
@@ -675,4 +676,102 @@ Only named replacements (:name) are allowed in literal() because we cannot guara
       });
     });
   });
+
+  describe('index hints', () => {
+    it('should add an index hint', () => {
+      expectsql(() => queryGenerator.selectQuery(User.table, {
+        model: User,
+        attributes: ['id'],
+        indexHints: [{ type: IndexHints.FORCE, values: ['index_project_on_name'] }],
+      }, User), {
+        default: buildInvalidOptionReceivedError('quoteTable', sequelize.dialect.name, ['indexHints']),
+        'mariadb mysql snowflake': 'SELECT [id] FROM [Users] AS [User] FORCE INDEX ([index_project_on_name]);',
+      });
+    });
+
+    it('should add an index hint with multiple values', () => {
+      expectsql(() => queryGenerator.selectQuery(User.table, {
+        model: User,
+        attributes: ['id'],
+        indexHints: [{ type: IndexHints.IGNORE, values: ['index_project_on_name', 'index_project_on_name_and_foo'] }],
+      }, User), {
+        default: buildInvalidOptionReceivedError('quoteTable', sequelize.dialect.name, ['indexHints']),
+        'mariadb mysql snowflake': 'SELECT [id] FROM [Users] AS [User] IGNORE INDEX ([index_project_on_name],[index_project_on_name_and_foo]);',
+      });
+    });
+
+    it('should not add an index hint if the type is not valid', () => {
+      expectsql(() => queryGenerator.selectQuery(User.table, {
+        model: User,
+        attributes: ['id'],
+        // @ts-expect-error -- we are testing invalid values
+        indexHints: [{ type: 'INVALID', values: ['index_project_on_name'] }],
+      }, User), {
+        default: buildInvalidOptionReceivedError('quoteTable', sequelize.dialect.name, ['indexHints']),
+        'mariadb mysql snowflake': 'SELECT [id] FROM [Users] AS [User];',
```

Comment:
Maybe we should throw an error on invalid indexHints?

Reply:
Good idea, updated now